x.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import argparse
+import glob
+import os
+import pathlib
+import re
+import subprocess
+import sys
+
+CMAKE_REQUIRE_VERSION = (3, 13, 0)
+SEMVER_REGEX = re.compile(
+    r"""
+        ^
+        (?P<major>0|[1-9]\d*)
+        \.
+        (?P<minor>0|[1-9]\d*)
+        \.
+        (?P<patch>0|[1-9]\d*)
+        (?:-(?P<prerelease>
+            (?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)
+            (?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*
+        ))?
+        (?:\+(?P<build>
+            [0-9a-zA-Z-]+
+            (?:\.[0-9a-zA-Z-]+)*
+        ))?
+        $
+    """,
+    re.VERBOSE,
+)
+
+def run(args, msg=None, **kwargs):
+    sys.stdout.flush()
+    p = subprocess.Popen(args, **kwargs)
+    code = p.wait()
+    if code != 0:
+        err = f"""
+failed to run: {args}
+exit with code: {code}
+error message: {msg}
+"""
+        raise RuntimeError(err)
+    else:
+        return p.stdout
+
+def find_command(command, msg=None):
+    output = run(["bash", "-c", f"command -v {command}"], stdout=subprocess.PIPE)
+    path = output.read().decode().strip()
+    run(["test", "-x", path], msg=msg)
+    return path
+
+def build(args):
+    (dir, jobs, ghproxy, ninja, unittest, compiler, d) = (args.dir, args.jobs, args.ghproxy, args.ninja, args.unittest, args.compiler, args.D)
+
+    basedir = pathlib.Path(__file__).parent.absolute()
+
+    find_command("autoconf", msg="autoconf is required to build jemalloc")
+    cmake = find_command("cmake", msg="CMake is required")
+
+    output = run([cmake, "-version"], stdout=subprocess.PIPE)
+    output = run(["head", "-n", "1"], stdin=output, stdout=subprocess.PIPE)
+    output = run(["sed", "s/[^0-9.]*//g"], stdin=output, stdout=subprocess.PIPE)
+    cmake_version = output.read().decode().strip()
+    cmake_require_version = '.'.join(map(str, CMAKE_REQUIRE_VERSION))
+    cmake_semver = SEMVER_REGEX.match(cmake_version)
+    if cmake_semver is None:
+        raise RuntimeError(f"CMake {cmake_require_version} or higher is required, got: {cmake_version}")
```

Comment:
I think we can provide a URL for users and developers to download cmake, like https://github.com/Kitware/CMake/releases or https://cmake.org/download/

Reply:
These can be done in subsequent PRs.

Reply:
I think it's too cumbersome to convey possible download links. Still we don't recommend for autoconf and users should be easily to search how to download CMake. For example, as a macOS user I'd prefer to manage CMake via Homebrew instead of download from either links provided here.

It's too much to a downstream project heuristically educating users to install a dependency.

Reply:
Also we already provide some preferred defaults in the README.md file.

Reply:
> Also, I think it would be good to provide a `--cmake-bin` to customize the use of the cmake executable.

Yes. This sounds reasonable to me.

Reply:
@PragmaTwice since this patch merged, you may create an issue for `--cmake-bin`.